pkg/apis/eventing/v1alpha1/trigger_lifecycle_test.go

```diff
@@ -329,14 +349,20 @@ func TestTriggerIsReady(t *testing.T) {
 			} else {
 				ts.MarkSubscriberResolvedFailed("Unable to get the Subscriber's URI", "subscriber not found")
 			}
-			if test.dependencyAnnotationExists && !test.dependencyStatusReady {
-				ts.MarkDependencyFailed("Dependency is not ready", "Dependency is not ready")
-			} else {
+			if !test.dependencyAnnotationExists {
 				ts.MarkDependencySucceeded()
+			} else {
+				if test.dependencyStatus == corev1.ConditionTrue {
+					ts.MarkDependencySucceeded()
+				} else if test.dependencyStatus == corev1.ConditionUnknown {
+					ts.MarkDependencyUnknown("The status of dependency is unknown", "The status of dependency is unknown: nil")
+				} else {
+					ts.MarkDependencyFailed("The status of dependency is false", "The status of dependency is unknown: nil")
```

Comment:
Yep

Reply:
That is only for batch type resources.

Reply:
What is the definition of "batch type resource"? I checked the codebase, for example https://github.com/knative/eventing/blob/master/pkg/apis/eventing/v1alpha1/broker_lifecycle.go, it use `MarkXXXXFailed` for `TriggerChannel`, `Ingress` and `Filter`.

Reply:
Think of it as 'run to completion', once you create a managed condition set, you either create a "living" or "batch". Living means it has a Top Level Condition that is Ready (True, False, or Unknown), creating a Batch means it has a Top Level Condition that is Succeeded (True, False, or Unknown). I think Batch is a poor name but I think it comes from there since k8s Batch resources have this condition for them.

Reply:
I checked k8s docs, https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/ is an example of batch type resources, right? In knative eventing, is there any resources that belongs to batch type resources. It looks like we should create another issue to fix this. I checked the code, knative eventing uses `MarkXXXFailed` for all resources even they belongs to "living".